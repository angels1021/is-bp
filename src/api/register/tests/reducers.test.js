/*
*
* Tests for register reducers
*
* */

import registerReducer from '../reducers';
import {
  changeForm,
  sendingRequest,
  requestError,
  clearError
} from '../actions';

describe('Reducers for register api', () => {
  let state;
  beforeEach(() => {
    state = registerReducer(undefined, {});
  });

  it('should return the initial state', () => {
    expect(state).toMatchSnapshot();
  });

  describe('changeForm', () => {
    it('should handle the changeForm action correctly', () => {
      // arrange
      const expectedResult = state
        .mergeIn(['form'], {
          username: 'Tom',
          password: 'McFly'
        });
      // act
      const result = registerReducer(state, changeForm({
        username: 'Tom',
        password: 'McFly'
      }));
      // assert
      expect(expectedResult).toEqual(result);
    });

    it('should handle the changeForm action correctly with partial data', () => {
      // arrange
      const expectedResult = state
        .mergeIn(['form'], {
          username: 'Tom',
          password: ''
        });
      // act
      const result = registerReducer(state, changeForm({
        username: 'Tom'
      }));
      // assert
      expect(expectedResult).toEqual(result);
    });
  });

  describe('sendingRequest', () => {
    it('should handle the sendingRequest action correctly', () => {
      // arrange
      const expectedResult = state.set('busy', true);
      // act
      const result = registerReducer(state, sendingRequest(true));
      // assert
      expect(expectedResult).toEqual(result);
    });
  });

  describe('requestError', () => {
    it('should handle the requestError action correctly', () => {
      // arrange
      const expectedResult = state.set('error', 'exception');
      // act
      const result = registerReducer(state, requestError('exception'));
      // assert
      expect(expectedResult).toEqual(result);
    });
  });

  describe('clearError', () => {
    it('should handle the clearError action correctly', () => {
      // arrange
      const expectedResult = state.set('error', false);
      // act
      const result = registerReducer(state, clearError());
      // assert
      expect(expectedResult).toEqual(result);
    });
  });
});
