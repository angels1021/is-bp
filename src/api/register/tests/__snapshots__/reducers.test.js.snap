// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducers for register api should return the initial state 1`] = `
Object {
  "busy": false,
  "error": false,
  "form": Object {
    "password": "",
    "username": "",
  },
}
`;
